core/src/Kokkos_CopyViews.hpp

```diff
@@ -3754,9 +3754,29 @@ create_mirror_view(const Space&, const Kokkos::View<T, P...>& src,
 }  // namespace Impl
 
 template <class T, class... P>
-typename Kokkos::View<T, P...>::HostMirror create_mirror_view(
-    Kokkos::View<T, P...> const& v) {
-  return Impl::create_mirror_view(v, Impl::ViewCtorProp<>{});
+std::enable_if_t<
+    std::is_same<
+        typename Kokkos::View<T, P...>::memory_space,
+        typename Kokkos::View<T, P...>::HostMirror::memory_space>::value &&
+        std::is_same<
+            typename Kokkos::View<T, P...>::data_type,
+            typename Kokkos::View<T, P...>::HostMirror::data_type>::value,
```

Comment:
Comment why this heinous code is needed for or (better) define some helper trait you can reuse with `DynRankView`
```C++
template <class>
struct has_sacado_specialization;
template <class T, class... P>
struct has_sacado_specialization<View<T, P...>> : std::integral_constant<PASTE HERE> {};
```

Reply:
This is the condition we were using before and are still using in the `Impl` overloads. I don't think naming it something with "sacado" in the name would be appropriate.

Reply:
Missing the second condition on the data type here already fails when only enabling `KokkosKernels` and its tests in `Trilinos`.